sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -2623,8 +2623,16 @@ object Sequence {
         // about a month length in days and a day length in microseconds
         val intervalStepInMicros =
```

Comment:
it looks like the step should not be physical seconds, but logical interval. cc @MaxGekk

Reply:
The current implementation is strange mix, it seems. There are following options:
1. Step is an interval of (months, days, micros):
    1. If start point is TimestampType, we should convert it to local date-time in the session time zone, and add the interval by time components. The intermediate local timestamps should be converted back to micros using the session time zone but we should keep adding the interval to local timestamp "accumulator".
    2. The same for dates - convert `start` to a local date. Time zone shouldn't involved here.

2. If the step is a duration in micros or days (this is not our case)
    1. start is TimestampType, we shouldn't convert it to local timestamp, and just add micros to instants. So, time zone will be not involved here.
    2. start is DateType, just add number of days. The same as for timestamps, time zone is not involved here.

Reply:
`start is DateType, just add number of days. The same as for timestamps, time zone is not involved here.` seems strange in different `tz`

Reply:
hi @cloud-fan，as  @MaxGekk  explain here, I am not sure if this patch look ok，I am willing to `add more documents to TemporalSequenceImpl`if we can follow this way rather than refactor a little.